### PR TITLE
SDI-554 Set audit module to DPS_SYNCHRONISATION for adjustments CRUD

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.repository.StoredProcedureRepository
+
+@Component
+@Aspect
+@Order(1)
+class AuditInterceptor(private val storedProcedureRepository: StoredProcedureRepository) {
+  @Around("@annotation(Audit)")
+  fun annotatedEndpoint(proceedingJoinPoint: ProceedingJoinPoint): Any? {
+    storedProcedureRepository.audit("DPS_SYNCHRONISATION")
+    try {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt:17]: if we ever need to this to be more "dynamic" we can move value to annotation

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import org.aspectj.lang.ProceedingJoinPoint
+import org.aspectj.lang.annotation.Around
+import org.aspectj.lang.annotation.Aspect
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.repository.StoredProcedureRepository
+
+@Component
+@Aspect
+@Order(1)
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt:12]: Why order 1 and not 0?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt:12]: I assumed 1 was the lowest number, and it fixed the actually transaction issue I predicted

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/audit/AuditInterceptor.kt:12]: BTW looks like any number below the default of Integer.MAX_VALUE would have worked

**Pull request merged**